src/createApiCall.ts

```diff
@@ -89,23 +100,38 @@ export function createApiCall(
 
         const streaming = (currentApiCaller as StreamingApiCaller).descriptor
           ?.streaming;
+
         const retry = thisSettings.retry;
-        if (
-          !streaming &&
-          retry &&
-          retry.retryCodes &&
-          retry.retryCodes.length > 0
-        ) {
-          retry.backoffSettings.initialRpcTimeoutMillis =
-            retry.backoffSettings.initialRpcTimeoutMillis ||
-            thisSettings.timeout;
-          return retryable(
-            func,
-            thisSettings.retry!,
-            thisSettings.otherArgs as GRPCCallOtherArgs,
-            thisSettings.apiName
+        if (!streaming && retry && retry.getResumptionRequestFn) {
+          throw new Error(
+            'Resumption strategy can only be used with server streaming retries'
           );
         }
+        if (!streaming && retry && retry.retryCodesOrShouldRetryFn) {
+          if (
+            retry.retryCodesOrShouldRetryFn instanceof Array &&
+            retry.retryCodesOrShouldRetryFn.length > 0
+          ) {
+            retry.backoffSettings.initialRpcTimeoutMillis =
+              retry.backoffSettings.initialRpcTimeoutMillis ||
+              thisSettings.timeout;
+            return retryable(
+              func,
+              thisSettings.retry!,
+              thisSettings.otherArgs as GRPCCallOtherArgs,
+              thisSettings.apiName
+            );
+          } else {
+            if (
+              retry.retryCodesOrShouldRetryFn instanceof Function &&
+              !streaming
+            ) {
+              throw new Error(
+                'Using a function to determine retry eligibility is only supported with server streaming calls'
+              );
+            }
+          }
```

Comment:
...or else?  maybe throw an error if it's neither an array nor a function

Reply:
Right now, this would happen because typescript would get mad if someone tried to pass in something that wasn't an array or a function. Given that, do you still want an explicit check?

Reply:
done - I went with the Sofia method because there are cases where it's okay for retryCodes to be empty - we are now instead failing faster